MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```diff
@@ -18,16 +18,29 @@ final class FeedViewModel {
 		Localized.Feed.title
 	}
 
-	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onStateChange: Observer<State>?
```

Comment:
The concept of `loading` and `error` should be separate since `errors` are not just for loading errors. We can represent errors for many other operations.

So ideally, there should be a separate observer for the error, as shown in the lectures.

```suggestion
	var onLoadingStateChange: Observer<Bool>?
	var onErrorStateChange: Observer<Bool>?
```

This way, you could have added this new feature without changing any code related to `loading`!

Reply:
this approach was inspired by mentoring session #008 for MVVM and RxSwift. 
I remember Caio saying it makes it much easier to combine streams into one stream, otherwise it gets sometimes out of hand.
In this session, there were no loading and errors but only the state of the form, maybe things that are related are advisable to be in one stream but things like loading and error where they are not related are in different streams. 
I am not sure whats your opinion is on this?

Reply:
Hi @Omar-Bassyouni! In session 008, I recommended keeping the *table state* (sections/rows state) in a single property. But unrelated functionality should be separated in other streams/observers because unrelated parts shouldn't affect others. For example, changing the error state shouldn't alter the table sections/rows - so they should be separate. In the same way, loading and error states are separate concerns. Nothing stops us from showing an error while still loading (and vice-versa!).

Reply:
ok, got it now.
Awesome thanks for the amazing PR 🚀